middleware/envcheckmiddleware.php

```diff
@@ -130,95 +136,41 @@ private function validateAndSetTokenBasedEnv() {
 		}
 	}
 
-	/**
-	 * Validates a token to make sure its linked to a valid resource
-	 *
-	 * Logic mostly duplicated from @see \OCA\Files_Sharing\Helper
-	 *
-	 * @fixme setIncognitoMode in 8.1 https://github.com/owncloud/core/pull/12912
-	 *
-	 * @param string $token
-	 *
-	 * @return array
-	 *
-	 * @throws CheckException
-	 */
-	private function getLinkItem($token) {
+    /**
+     * Validates a token to make sure its linked to a valid resource
+     *
+     * Uses Share 2.0
+     *
+     * @fixme setIncognitoMode in 8.1 https://github.com/owncloud/core/pull/12912
+     *
+     * @param string $token
+     * @return NotFoundResponse|Share\IShare
+     */
+    private function getLinkItem($token) {
 		// Allows a logged in user to access public links
 		\OC_User::setIncognitoMode(true);
 
-		$linkItem = Share::getShareByToken($token, false);
-
-		$this->checkLinkItemExists($linkItem);
-		$this->checkLinkItemIsValid($linkItem, $token);
-		$this->checkItemType($linkItem);
+        try {
+            $linkItem = $this->shareManager->getShareByToken($token);
+        } catch (ShareNotFound $e) {
+            return new NotFoundResponse();
```

Comment:
Well this is partly true.

* The sharemanager only handles shares of type file and folder. (So that check can go). (`checkLinkItemExists`)
* I actually don't get the use of `checkItemType` (`checkItemType` seems to already take care of it).
* `checkLinkItemIsValid` the share manager and share object take care of this.

Basically we have lazy shares. So we set all the stuff in the share object. But we actually do not check if the node exists until you do $share->getNode.